src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -1,654 +1,650 @@
-
-    /**
-     * Updates the equipment. Always take the first item of each type and set it.
+     * Decide what we should do next!
+     *
+     * @return the next AIState.
      */
-    protected void updateArmor()
+    protected AIState decide()
     {
-        worker.setItemStackToSlot(EntityEquipmentSlot.CHEST, ItemStackUtils.EMPTY);
-        worker.setItemStackToSlot(EntityEquipmentSlot.FEET, ItemStackUtils.EMPTY);
-        worker.setItemStackToSlot(EntityEquipmentSlot.HEAD, ItemStackUtils.EMPTY);
-        worker.setItemStackToSlot(EntityEquipmentSlot.LEGS, ItemStackUtils.EMPTY);
-
-        for (int i = 0; i < new InvWrapper(worker.getInventoryCitizen()).getSlots(); i++)
+        setDelay(20);
+        for (final ToolType toolType : toolsNeeded)
         {
-            final ItemStack stack = worker.getInventoryCitizen().getStackInSlot(i);
-
-            if (ItemStackUtils.isEmpty(stack))
+            if (getOwnBuilding() != null && !InventoryUtils.hasItemHandlerToolWithLevel(new InvWrapper(getInventory()),
+              toolType,
+              0,
+              getOwnBuilding().getMaxToolLevel()))
             {
-                new InvWrapper(worker.getInventoryCitizen()).extractItem(i, Integer.MAX_VALUE, false);
-                continue;
+                return START_WORKING;
             }
+        }
 
-            if (stack.getItem() instanceof ItemArmor && worker.getItemStackFromSlot(((ItemArmor) stack.getItem()).armorType) == ItemStackUtils.EMPTY)
+        if (worker.getWorkBuilding() != null
+              && !(worker.getLastAttackedEntity() != null
+              && !worker.getLastAttackedEntity().isDead)
+              && getOwnBuilding() instanceof AbstractBuildingGuards
+              && target == null)
+        {
+            final AbstractBuildingGuards guardBuilding = (AbstractBuildingGuards) getOwnBuilding();
+
+            switch (guardBuilding.getTask())
             {
-                worker.setItemStackToSlot(((ItemArmor) stack.getItem()).armorType, stack);
+                case PATROL:
+                    if (currentPatrolPoint == null)
+                    {
+                        currentPatrolPoint = guardBuilding.getNextPatrolTarget(currentPatrolPoint);
+                    }
+                    if (currentPatrolPoint != null
+                      && worker.isWorkerAtSiteWithMove(currentPatrolPoint, 1))
+                    {
+                        currentPatrolPoint = guardBuilding.getNextPatrolTarget(currentPatrolPoint);
+                    }
+                    break;
+                case GUARD:
+                    worker.isWorkerAtSiteWithMove(guardBuilding.getGuardPos(), GUARD_POS_RANGE);
+                    break;
+                case FOLLOW:
+                    worker.isWorkerAtSiteWithMove(guardBuilding.getPlayerToFollow(), GUARD_POS_RANGE);
+                    break;
+                default:
+                    worker.isWorkerAtSiteWithMove(worker.getWorkBuilding().getLocation(), 10);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 10 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)